src/hotspot/cpu/x86/x86.ad

```diff
@@ -3682,7 +3682,7 @@ instruct vconvF2HF_mem_reg(memory mem, vec src) %{
   ins_pipe( pipe_slow );
 %}
 
-instruct convHF2F_reg_reg(regF dst, rRegI src) %{
+instruct convHF2F_reg_reg(vlRegF dst, rRegI src) %{
```

Comment:
I think we can remove change from destination operand, it will automatically be assigned register from lower register bank for non-evex targets based on dynamic_register_classes. Problem is manly around incorrect register class for source operands since their definitions may not be constrained to allocate from lower register bank unless we use a legacy register class (vlRegF).